scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -160,41 +189,113 @@
                                  [:expected/starting :missing] (launch-task api-client expected-state-dict)
                                  [:expected/starting :pod/running] (pod-has-started kcc existing-state-dict)
-                                 ; TODO We need to add cases for the other [:expected/starting *] states in.
-                                 [:expected/starting :pod/waiting] expected-state-dict
-                                 [:expected/starting :pod/succeeded] (pod-has-just-completed kcc existing-state-dict)
-                                 [:expected/starting :pod/failed] (pod-has-just-completed kcc existing-state-dict)
+                                 [:expected/starting :pod/waiting] expected-state-dict ; Its starting. Can be stuck here.
+                                 [:expected/starting :pod/succeeded] (pod-has-just-completed kcc existing-state-dict) ; Finished fast.
+                                 [:expected/starting :pod/failed] (pod-has-just-completed kcc existing-state-dict) ; TODO: May need to mark mea culpa retry
                                  [:expected/running :pod/running] expected-state-dict
                                  [:expected/running :pod/succeeded] (pod-has-just-completed kcc existing-state-dict)
-                                 [:expected/completed :pod/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                                 [:expected/running :pod/failed] (pod-has-just-completed kcc existing-state-dict)
-                                 [:expected/completed :pod/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
+
+                                 ; This is an exception. The writeback to datomic has occurred, so there's nothing to do except to delete the task from kubernetes
+                                 ; and remove it from our tracking.
+                                 [:expected/completed :pod/succeeded] (delete-task api-client pod)
+
+                                 [:expected/running :pod/failed] (pod-has-just-completed kcc existing-state-dict) ; TODO: May need to mark mea culpa retry
+                                 [:expected/running :pod/waiting] (do ; This case is weird.
+                                                                    ; This breaks our rule of calling pod-has-completed on a non-terminal pod state.
+                                                                    (kill-task-weird api-client expected-state-dict existing-state-dict)
+                                                                    (pod-has-just-completed kcc existing-state-dict)) ; TODO: Should mark mea culpa retry
+
+                                 [:expected/completed :pod/failed] (delete-task api-client pod)
                                  [:expected/completed :missing] nil ; Cause it to be deleted.
```

Comment:
What does the `it` in the `Cause it to be deleted refer to`. If some external actor is performing this state cleanup, can we mention it by name here?

Reply:
> removes it from

What is `it` referring to?

Reply:
Fixed.